### PR TITLE
add helm repo creds and module application Helm

### PR DESCRIPTION
**Pull request by 0xbase12**
- add helm repo creds
- add module application Helm

Closes https://github.com/SixSq/tasklist/issues/2981
Closes https://github.com/SixSq/tasklist/issues/2982
Closes https://github.com/SixSq/tasklist/issues/2987

Recreated from merged PR by mistake. Sorry @konstan

### CODE REVIEW

**Comment by konstan** [comment]
hm... @0xbase12 I can't assign the review of this PR to you because you created the PR :/ 

So, anyways... the PR is ready for review. @0xbase12 please review it. Thanks.

File: code/src/sixsq/nuvla/server/resources/spec/module_application_helm.cljc [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+             :json-schema/description "Helm chart absolute URL")))
+
+
+(s/def ::helm-chart-version
+  (-> (st/spec ::core/nonblank-string)
+      (assoc :name "helm-chart-version"
+             :json-schema/description "Helm chart version to deploy, latest (if empty)")))
+
+
+(s/def ::helm-repo-creds
```
Comment (0xbase12) [inline code/src/sixsq/nuvla/server/resources/spec/module_application_helm.cljc:38]: Is this a credential that allow to access the helm registry?

File: code/src/sixsq/nuvla/server/resources/module_application_helm.clj [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+            helm-absolute-url]} :body :as request}]
+  (if (and (or (nil? helm-repo-url) (nil? helm-chart-name))
+          (nil? helm-absolute-url))
+      (throw (r/ex-response
+               "Mandatory fields missing: (:helm-repo-url and :helm-chart-name) or :helm-absolute-url"
+               400))
+      request))
+
+
+(defn- trow-mandatory-fields-mismatch
```
[the remark this answers is not in the document]
Comment (konstan) [reply code/src/sixsq/nuvla/server/resources/module_application_helm.clj]: fixed

File: code/src/sixsq/nuvla/server/resources/deployment/utils.clj [hunk 3]
```diff
@@ -236,15 +236,18 @@
                            (some->> deployment :registries-credentials
                                     (map crud/retrieve-by-id-as-admin)))
         registries-infra (when full
-                           (map (comp crud/retrieve-by-id-as-admin :parent) registries-creds))]
+                           (map (comp crud/retrieve-by-id-as-admin :parent) registries-creds))
+        helm-repo-cred   (some-> (-> deployment :module :content :helm-repo-creds)
```
[the remark this answers is not in the document]
Comment (konstan) [reply code/src/sixsq/nuvla/server/resources/deployment/utils.clj]: yes - fixed

File: code/src/sixsq/nuvla/server/resources/spec/module_application_helm.cljc (hunk 1 above)
Comment (konstan) [reply code/src/sixsq/nuvla/server/resources/spec/module_application_helm.cljc:38]: Yes. I updated the docu string.

**Pull request merged**